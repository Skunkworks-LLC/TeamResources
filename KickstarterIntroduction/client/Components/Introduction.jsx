import NavBar from './navbar.jsx';
import Header from './header.jsx';
import Player from './player.jsx';
import Status from './status.jsx';
import OwnerModal from './modals/ownermodal.jsx';
import BackProjectModal from './modals/backprojectmodal.jsx';
import OtherProjectsModal from './modals/otherprojectsmodal.jsx';
import FollowProjectModal from './modals/followprojectmodal.jsx';
import axios from 'axios';
import React from 'react';
import ReactDOM from 'react-dom';
import './styles/introduction.css';
import './styles/Fonts/MaisonNeue/fonts.css';

class Introduction extends React.Component {
  constructor(props){
    super(props);
    this.state = {
      project: {},
      owner: {},
      player: {},
      status: {},
      displayPledge: false,
      displayOwner: false,
      displayOtherProjects: false,
      displayFollow: false,

    }
    this.getProject = this.getProject.bind(this);
    this.ownerClicked = this.ownerClicked.bind(this);
    this.backButtonClicked = this.backButtonClicked.bind(this);
    this.otherProjectsClicked = this.otherProjectsClicked.bind(this);
    this.followProjectClicked = this.followProjectClicked.bind(this);
  }

//Esc key can be pressed to set all displays to false (to exit modal using this key)
  componentDidMount() {
    this.getProject();
    window.addEventListener('keydown', event => {
      if (event.keyCode === 27){
        this.setState({
          displayOwner: false,
          displayPledge: false,
          displayOtherProjects: false,
          displayFollow: false,
        })
      }
    })
  }

  getProject(projectid = window.location.pathname.slice(1)) {
    //Set to 1 for now, but project id can be passed in to fetch specific project
    axios.get(`http://127.0.0.1:3000/projects/${projectid}`)
    .then(project => {
      this.setState({
        project: project.data[0],
        owner: project.data[0].owner,
        player: project.data[0].player,
        status: {
          currentRaised: project.data[0].status.currentRaised,
          targetGoal: project.data[0].status.targetGoal,
          backers: project.data[0].status.backers.toLocaleString(),
          currentRaisedString: project.data[0].status.currentRaised.toLocaleString(),
          deadline: project.data[0].status.deadline,
          targetGoalString: project.data[0].status.targetGoal.toLocaleString(),
        },
      })
    })
  }

//Lines 69 through 80 are the click handlers to set corresponding modal display to true
  ownerClicked() {
    this.setState({displayOwner: !this.state.displayOwner});
  }
  backButtonClicked() {
    this.setState({displayPledge: !this.state.displayPledge});
  }
  otherProjectsClicked() {
    this.setState({displayOtherProjects: !this.state.displayOtherProjects});
  }
  followProjectClicked() {
    this.setState({displayFollow: !this.state.displayFollow});
  }

//Lines 94 through 105 will render the corresponding modal if its state is set to true.
  render() {
    return (
      <div>
        <div id="navigation-bar">
          <NavBar />
        </div>
        <div id="header">
          <Header project={this.state.project} owner={this.state.owner} ownerClicked={this.ownerClicked} otherProjectsClicked={this.otherProjectsClicked}/>
        </div>
          {this.state.displayOwner ? 
          <OwnerModal ownerClicked={this.ownerClicked} owner={this.state.owner} player={this.state.player}/>
          : null}
          {this.state.displayOtherProjects ? 
          <OtherProjectsModal owner={this.state.owner} otherProjectsClicked={this.otherProjectsClicked}/>
          : null}
          {this.state.displayPledge ?
          <BackProjectModal backButtonClicked={this.backButtonClicked} project={this.state.project} owner={this.state.owner}/>
          : null}
          {this.state.displayFollow ? 
          <FollowProjectModal followProjectClicked={this.followProjectClicked} />
          : null}
        <div className="player-and-status-container">
          <Player project={this.state.project} player={this.state.player}/><Status stats={this.state.status} backButtonClicked={this.backButtonClicked} followProjectClicked={this.followProjectClicked}/>
        </div>
      </div>
    )
  }
}

export default Introduction;