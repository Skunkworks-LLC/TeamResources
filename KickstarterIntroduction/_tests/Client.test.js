import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Introduction from '../client/Components/introduction.jsx';
import NavBar from '../client/Components/navbar.jsx';
import Header from '../client/Components/header.jsx';
import Player from '../client/Components/player.jsx';
import Status from '../client/Components/status.jsx';
import BackProjectModal from '../client/Components/backprojectmodal.jsx';
import OwnerModal from '../client/Components/ownermodal.jsx';

Enzyme.configure({adapter: new Adapter()});

describe('<Introduction />', () => {
  describe('render()', () => {
    test('renders the main introduction component', () => {
      const wrapper = shallow(<Introduction />);

      expect(toJson(wrapper)).toMatchSnapshot();
    });
  });
});

describe('<NavBar />', () => {
  describe('render()', () => {
    test('renders the component for NavBar', () => {
      const wrapper = shallow(<NavBar />);

      expect(toJson(wrapper)).toMatchSnapshot();
    });
  });
});

describe('<Header />', () => {
  describe('render()', () => {
    test('renders the component for header', () => {
      const ownerObj = {avatar: 'url here'}
      const projectObj = {name: 'something'}
      const wrapper = shallow(<Header owner={ownerObj} project={projectObj}/>);

      expect(toJson(wrapper)).toMatchSnapshot();
    });
  });
});

describe('<BackProjectModal />', () => {
  describe('render()', () => {
    test('renders the component for header', () => {
      const ownerObj = {name: 'Jerry'};
      const projectObj = {name: 'Hahaha'};
      const wrapper = shallow(<BackProjectModal project={projectObj} owner={ownerObj}/>);

      expect(toJson(wrapper)).toMatchSnapshot();
    });
  });
});

describe('<OwnerModal />', () => {
  describe('render()', () => {
    test('renders the component for header', () => {
      const ownerObj = {name: 'Jerry'};
      const playerObj={source: 'http://www.youtube.com/', location: 'Athens, GA'}
      const wrapper = shallow(<OwnerModal owner={ownerObj} player={playerObj}/>);

      expect(toJson(wrapper)).toMatchSnapshot();
    });
  });
});

describe('<Player />', () => {
  describe('render()', () => {
    test('renders the component for player', () => {
      const playerObj={source: 'http://www.youtube.com/', location: 'Athens, GA'}
      const wrapper = shallow(<Player player={playerObj}/>);

      expect(toJson(wrapper)).toMatchSnapshot();
    });
  });
});

describe('<Status />', () => {
  describe('render()', () => {
    test('renders the component for status', () => {
      const statsObj={currentRaised: 505, targetGoal: 1000}
      const wrapper = shallow(<Status stats={statsObj}/>);

      expect(toJson(wrapper)).toMatchSnapshot();
    });
  });
});
