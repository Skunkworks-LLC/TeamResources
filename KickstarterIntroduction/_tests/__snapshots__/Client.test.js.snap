// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BackProjectModal /> render() renders the component for header 1`] = `
<div>
  <div
    className="back-project-modal"
  >
    <div
      className="back-project-form"
    >
      <div
        className="back-project-form-header"
      >
        <button
          className="close-project-modal"
          onClick={[Function]}
        >
          ×
        </button>
        <img
          height="18.9375"
          src="https://d3mlfyygrfdi2i.cloudfront.net/kickstarter-logo-color.png"
          width="185"
        />
      </div>
      <div
        className="back-project-title"
      >
        <h1>
          Hahaha
        </h1>
        <h5>
          By 
          Jerry
        </h5>
      </div>
      <div
        className="back-project-form-main"
      >
        <div
          className="back-project-form-pledges"
        >
          <h3>
            Support this project
          </h3>
          <ul
            className="list-project-pledges"
          >
            <li
              className="list-project-pledge-item"
            >
              Pledge/support module can go here
            </li>
            <li
              className="list-project-pledge-item"
            >
              Pledge/support module can go here
            </li>
            <li
              className="list-project-pledge-item"
            >
              Pledge/support module can go here
            </li>
            <li
              className="list-project-pledge-item"
            >
              Pledge/support module can go here
            </li>
            <li
              className="list-project-pledge-item"
            >
              Pledge/support module can go here
            </li>
          </ul>
        </div>
        <div
          className="back-project-form-info"
        >
          <h5>
            <b>
              Kickstarter is not a store.
            </b>
          </h5>
          <h5>
            It's a way to bring creative projects to life.
          </h5>
          <p>
            Kickstarter does not guarantee projects or investigate a creator's ability to complete their project. It is the responsibility of the project creator to complete their project as promised, and the claims of this project are theirs alone.
          </p>
          <br />
          <p>
            Learn more about accountability
          </p>
          <br />
          <h5>
            FREQUENTLY ASKED QUESTIONS
          </h5>
          <ul
            className="frequently-asked-questions"
          >
            <li
              className="faq-question"
            >
              ► How do I pledge?
            </li>
            <li
              className="faq-question"
            >
              ► When is my card charged?
            </li>
            <li
              className="faq-question"
            >
              ► So I'm only charged if funding succeeds?
            </li>
            <li
              className="faq-question"
            >
              ► What can others see about my pledge?
            </li>
            <li
              className="faq-question"
            >
              ► What if I want to change my pledge?
            </li>
            <li
              className="faq-question"
            >
              ► If this project is funded, how do I get my reward?
            </li>
          </ul>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<Header /> render() renders the component for header 1`] = `
<div
  className="header-container"
>
  <div
    className="header-owner"
  >
    <h1
      onClick={[Function]}
    >
      <img
        className="avatar"
        src="url here"
      />
    </h1>
    <div
      id="author"
    >
      <a
        className="black-link"
        href="#"
        onClick={[Function]}
      >
        By 
      </a>
      <br />
      <a
        className="gray-link"
        href="#"
      >
         created
      </a>
    </div>
  </div>
  <div
    className="header-title"
  >
    <h1>
      something
    </h1>
    <h3
      id="project-tagline"
    />
  </div>
</div>
`;

exports[`<Introduction /> render() renders the main introduction component 1`] = `
<div
  onKeyDown={[Function]}
>
  <div
    id="navigation-bar"
  >
    <NavBar />
  </div>
  <div
    id="header"
  >
    <Header
      owner={Object {}}
      ownerClicked={[Function]}
      project={Object {}}
    />
  </div>
  <div
    className="player-and-status-container"
  >
    <Player
      player={Object {}}
      project={Object {}}
    />
    <Status
      backButtonClicked={[Function]}
      stats={Object {}}
    />
  </div>
</div>
`;

exports[`<NavBar /> render() renders the component for NavBar 1`] = `
<div
  className="main-navigation-bar"
>
  <div
    id="nav-bar-1"
  >
    <a
      className="black-link"
      href="#"
    >
      Explore
    </a>
    <a
      className="black-link"
      href="#"
    >
      Start a project
    </a>
  </div>
  <div
    id="nav-bar-2"
  >
    <img
      height="18.9375"
      src="https://d3mlfyygrfdi2i.cloudfront.net/kickstarter-logo-color.png"
      width="185"
    />
  </div>
  <div
    id="nav-bar-3"
  >
    <a
      className="black-link"
      href="#"
      onClick={[Function]}
    >
      Search 
      <i
        className="fas fa-search"
      />
    </a>
    <a
      className="black-link"
      href="#"
    >
      Sign in
    </a>
  </div>
</div>
`;

exports[`<OwnerModal /> render() renders the component for header 1`] = `
<div>
  <div
    className="owner-modal"
  >
    <div
      className="owner-modal-box"
    >
      <button
        className="close-project-modal"
        onClick={[Function]}
      >
        ×
      </button>
      <center>
        <h4>
          About the creator
        </h4>
      </center>
      <br />
      <h1>
        Jerry
      </h1>
      <h6>
        Athens, GA
      </h6>
      <p
        className="owner-summary"
      />
    </div>
  </div>
</div>
`;

exports[`<Player /> render() renders the component for player 1`] = `
<div
  className="player-box"
>
  <iframe
    frameBorder="0"
    height="440"
    src="http://www.youtube.com/"
    width="740"
  />
  <br />
  <div
    className="under-player-buttons"
  >
    <a
      className="gray-link"
      href="#"
    >
      <i
        className="fab fa-kickstarter playlink"
      />
      Project We Love
    </a>
    <a
      className="gray-link"
      href="#"
    >
      <i
        className="far fa-compass playlink"
      />
      Publishing
    </a>
    <a
      className="gray-link"
      href="#"
    >
      <i
        className="fas fa-map-marker-alt playlink"
      />
      Athens, GA
    </a>
  </div>
</div>
`;

exports[`<Status /> render() renders the component for status 1`] = `
<div
  className="status-box"
>
  <progress
    max={1000}
    value={505}
  />
  <br />
  <div
    id="funds-raised"
  >
    $
  </div>
  <div
    className="project-stats-text"
  >
    pledged as of $
     goal
  </div>
  <br />
  <div
    className="project-stats"
  />
  <div
    className="project-stats-text"
  >
    backers
  </div>
  <br />
  <div
    className="project-stats"
  >
    0
  </div>
  <div
    className="project-stats-text"
  >
    days to go
  </div>
  <br />
  <button
    className="status"
    id="back-project"
    onClick={[Function]}
  >
    Back this project
  </button>
  <br />
  <button
    className="status"
    id="follow-project"
    onClick={[Function]}
  >
    <div
      className="glyphicon glyphicon-bookmark"
    />
     Follow Project
  </button>
  <a
    className="gray-link"
    href="#"
  >
    <i
      className="fab fa-facebook"
    />
  </a>
  <a
    className="gray-link"
    href="#"
  >
    <i
      className="fab fa-twitter"
    />
  </a>
  <a
    className="gray-link"
    href="#"
  >
    <i
      className="fas fa-envelope"
    />
  </a>
  <a
    className="gray-link"
    href="#"
  >
    <i
      className="fas fa-code"
    />
  </a>
  <br />
  <br />
  <p
    id="project-term"
  >
    <u>
      All or nothing.
    </u>
    This project will only be funded if it reaches its goal by 
    Friday, September 14, 2018 12:08 PM
    .
  </p>
</div>
`;
